optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml

```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <persistence xmlns="http://java.sun.com/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              version="2.0"  xsi:schemaLocation="http://java.sun.com/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_0.xsd">
-  <persistence-unit name="optaplanner-persistence-jpa-test" transaction-type="JTA">
-    <provider>org.hibernate.ejb.HibernatePersistence</provider>
+  <persistence-unit name="org.optaplanner.persistence.jpa" transaction-type="JTA">
+    <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
```

Comment:
1) For my info, what's the difference between the ejb and jpa hibernate persistence provider?
2) Why do we configure one to begin with? It is related to the database sql dialect detection?

Reply:
Oh I missed this. org.hibernate.ejb.HibernateProvider is deprecated. @domhanak why was this change needed? Or is it just a change copied from jBPM changes? Please try if the changes work leaving the original provider there.

Reply:
@ge0ffrey 
1. From what I found - One is Hibernate persistence provider implementation and the other is Hibernate EJB3 persistence provider implementation.

@baldimir 
Reason i changed this is that org.hibernate.ejb.HibernatePersistence is deprecated  - https://docs.jboss.org/hibernate/orm/5.1/javadocs/org/hibernate/ejb/HibernatePersistence.html
Furthermore, org.hibernate.ejb.HibernateProvider is not part of newer versions of hibernate.
https://docs.jboss.org/hibernate/orm/5.2/javadocs/org/hibernate/ejb/HibernatePersistence.html

2. It is invoked by the Persistence class in Java SE environments to create an EntityManagerFactory. SQL Dialect is handled by dialect classes and is DB specific.